### PR TITLE
:seedling: Runtime sdk/flaky e2e fix

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon [kmuldoon@vmware.com](mailto:kmuldoon@vmware.com)

This PR updates the blocking e2e tests added in #6761 to use RetryAfterSeconds. This avoids triggering an exponential backoff in the reconciler and may help to resolve some flakes in the current tests.

### CODE REVIEW

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (sbueringer) [inline test/e2e/cluster_upgrade_runtimesdk.go:490]: I think we have to be careful with the error message given that we will hit this case if either:
* the hook is still blocking or
* (create failed | upgrade failed | delete failed)

This might become confusing for folks trying to debug a failed test run

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (ykakarap) [inline test/e2e/cluster_upgrade_runtimesdk.go:524]: Not relevant to this PR but just FYI incase you want t fix it: The comment on `framework.DeleteCluster` is misleading. The current comment suggests that it waits for cluster owned objects to be gone.

**Review by ykakarap — COMMENTED** [review]
I know this is a WIP. Doing just a high level review. The overall approach looks good. 

Will do another review once reaedy.

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (sbueringer) [inline test/e2e/cluster_upgrade_runtimesdk.go:494]: nit: Those statements were a little bit easier to read before (of course it works on big screens...)

**Comment by sbueringer** [comment]
>  I'm going to run it a number of times to make sure this fixes the flake as it's pretty disruptive on PRs at this point.
Thx!

Of course it confuses folks anyway, but this job is optional for merge (exactly because we weren't sure everything (CC+RuntimeSDK) is 100% stable yet)

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (killianmuldoon) [reply test/e2e/cluster_upgrade_runtimesdk.go:380]: No - here we get a double rollout at present  so condition is unstable (rollout takes precedence over lifecycleHook blocked) I'm trying to figure out a way to block the rollout though.

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (killianmuldoon) [reply test/e2e/cluster_upgrade_runtimesdk.go:494]: Yeah - it's just a lot of vertical space for repetitive key values with most of these input structs. I'm fine either way though.

Comment (sbueringer) [reply test/e2e/cluster_upgrade_runtimesdk.go:494]: Let's say there are folks which have a hard-limit somewhere between 100 and 120 character's. 
So I personally try not to go over ~140 or something :P 

Given that we consistently do it multi-line across the entire e2e tests / test framework, I personally would use multi-line here as well.

I won't block the PR because of that though

Comment (sbueringer) [reply test/e2e/cluster_upgrade_runtimesdk.go:494]: (disclaimer, edited the comment multiple times)

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (fabriziopandini) [inline test/e2e/cluster_upgrade_runtimesdk.go:456]: Do we really need to wait 1 minute to confirm we are blocked? 30 seems enough to me...

Comment (killianmuldoon) [reply test/e2e/cluster_upgrade_runtimesdk.go:456]: For now this is mainly up to a minute because the condition test doesn't work for beforeClusterUpgrade (working on fixing this in this PR or a follow up) once that's fixed I think 30 seconds is enough, but and upgrade specifically could take long enough to kick in that I'm more comfortable with 60 seconds with the current test structure.

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (ykakarap) [inline test/e2e/cluster_upgrade_runtimesdk.go:476]: I prefer the original text. I believe this message only shows if the check fails and the original message showed why the check failed and why.

**Review by ykakarap — COMMENTED** [review]
lgtm pending the open comments

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (killianmuldoon) [reply test/e2e/cluster_upgrade_runtimesdk.go:476]: The issue is that the more specific message could be wrong sometimes. We don't know at this point if the hook failed to unblock or if reconciliation failed in some other way, so adding the detail makes this message kind of misleading.

Comment (sbueringer) [reply test/e2e/cluster_upgrade_runtimesdk.go:478]: Makes sense. I would also prefer a more generic message over a misleading one

**Comment by sbueringer** [comment]
Thx!
/lgtm

File: test/e2e/cluster_upgrade_runtimesdk.go
Comment (ykakarap) [reply test/e2e/cluster_upgrade_runtimesdk.go:474]: Sounds good. I think we can keep the messaging.

**Review by ykakarap — APPROVED** [review]
/lgtm

**Pull request merged**
